File: src/lib/logger.ts
import { invoke } from "@tauri-apps/api/core";

const isTauri = () =>
  typeof window !== "undefined" && "__TAURI_INTERNALS__" in window;

async function log(cmd: string, message: string, context?: any) {
  console.log(`[${cmd}]`, message, context ?? "");

  if (!isTauri()) return;

  try {
    await invoke(cmd, { message });
  } catch {}
}

export const logger = {
  debug: (message: string, context?: any) => log("log_debug", message),
  info: (message: string, context?: any) => log("log_info", message),
  warn: (message: string, context?: any) => log("log_warn", message),
  error: (message: string, context?: any) => log("log_error", message),
};

File: src-tauri/src/utils/logger.rs
use log;

#[tauri::command]
pub fn log_debug(message: String) {
    log::debug!("[React] {}", message);
}

#[tauri::command]
pub fn log_info(message: String) {
    log::info!("[React] {}", message);
}

#[tauri::command]
pub fn log_warn(message: String) {
    log::warn!("[React] {}", message);
}

#[tauri::command]
pub fn log_error(message: String) {
    log::error!("[React] {}", message);
}
